service/controller/resource/app/desired.go

```diff
@@ -194,20 +194,26 @@ func (r *Resource) newAppSpecs(ctx context.Context, cr apiv1alpha2.Cluster) ([]k
 		return nil, microerror.Mask(err)
 	}
 
-	apps, err := r.releaseVersion.AppVersion(ctx, &cr)
+	apps, err := r.releaseVersion.Apps(ctx, &cr)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
 	var specs []key.AppSpec
-	for appName, appVersion := range apps {
+	for appName, app := range apps {
+		var catalog string
+		if app.Catalog == "" {
```

Comment:
This is necessary because old release CRs might not have a populated `catalog` field.